### PR TITLE
Cleanup redundant code, type-check middleware, simplify import

### PR DESCRIPTION
**Pull request by mike-north**

### CODE REVIEW

File: lib/attach-middleware.js [hunk 1]
```diff
@@ -25,22 +24,12 @@ function installRouter(app) {
...
-};
-
-/**
- * Used when app is in ci mode (`ember test`).
- * Collects the coverage on each request and merges it into the coverage map.
- *
- * @param {express.Application} app
- */
-const testMiddleware = (app, _options) => {
+const middleware = (app, _options) => {
```
Comment (rwjblue) [inline lib/attach-middleware.js]: Its not immediately obvious why we need this function at all?

Wouldn't it be the same to replace callers with `installRouter`?

What am I missing?

File: lib/routes/update-snapshot.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+/** @type {e.RequestHandler} */
+const updateSnapshot = (request, response) => {
+  const {
+    body,
+    params: { modulename = 'default-module', testname, snapname }
+  } = request;
+  let errored = false;
+
+  const opts = {
+    show: Boolean(process.env.SNAPSHOT_SHOW),
```
Comment (rwjblue) [inline lib/routes/update-snapshot.js]: Should we extract this to another module (and fixup the sanitization)? Perhaps in a follow up PR though...

File: package.json [hunk 3]
```diff
@@ -36,7 +36,7 @@
     "ember-cli-typescript": "2.0.0-beta.3",
     "fs-extra": "7.0.0",
     "json-typescript": "1.0.1",
-    "qunit-snapshot": "1.2.1",
+    "qunit-snapshot": "1.3.0",
```
Comment (rwjblue) [inline package.json]: Any reason to lock this down (and prevent bug fixes from trickling down to users)?

Comment (mike-north) [reply package.json]: Still experimenting with @renovate-bot -- this is what it does by default

File: addon-test-support/index.ts [hunk 4]
```diff
@@ -24,7 +24,26 @@ interface SnapshotFile {
...
 
+/**
+ * Setup snapshot testing
+ * @param qunit QUnit global
+ * @public
+ * @deprecated
+ */
 export function install(qunit: QUnit = QUnit) {
+  // tslint:disable-next-line:no-console
+  console.warn(
```
[the remark this answers is not in the document]
Comment (mike-north) [reply addon-test-support/index.ts]: Fixed 1288dca

File: package.json (hunk 3 above)
Comment (mike-north) [reply package.json]: Fixed 4b6860f

File: lib/attach-middleware.js (hunk 1 above)
Comment (mike-north) [reply lib/attach-middleware.js]: Fixed 2934108

File: lib/routes/update-snapshot.js (hunk 2 above)
Comment (mike-north) [reply lib/routes/update-snapshot.js]: addressed here 876d75f

File: addon-test-support/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (rwjblue) [reply addon-test-support/index.ts]: > Do ember api docs need a fix?

Yes, apparently they do. Importing from `@ember/debug` has significant benefits (ember-cli-babel ensures that any `deprecate` calls are stripped from prod builds, this is not true of `@ember/application/deprecate`).

[the remark this answers is not in the document]
Comment (mike-north) [reply addon-test-support/index.ts]: filed https://github.com/emberjs/ember.js/issues/17185

**Pull request merged**